### PR TITLE
OC-1643 Show admin link from question in Studio.

### PR DESCRIPTION
**Pull request by jbzdak**
Testing instructions on: https://github.com/open-craft/xblock-dalite/pull/3

### CODE REVIEW

File: peerinst/templates/peerinst/question_list.html [hunk 1]
```diff
@@ -7,7 +7,7 @@
 {% for obj in object_list %}
   <li><a href="{% url 'question' assignment_id=assignment.pk question_id=obj.pk %}">
     {{ obj.pk }}
-  <a></li>
+  </a></li>
```
Comment (jbzdak) [inline peerinst/templates/peerinst/question_list.html]: This change is unrelated, just PyCharm warned me of malformed HTML.

File: peerinst/context_processors.py [hunk 2]
```diff
@@ -0,0 +1,11 @@
+
+def is_staff_user_from_lms(request):
```
Comment (pomegranited) [inline peerinst/context_processors.py]: I think this method should be called `is_staff_user_from_cms` or `is_staff_user_from_studio` instead of `_lms`, since it sets the context property to true when the staff user is in the CMS/Studio, rather than in the LMS.

File: dalite/tests.py [hunk 3]
```diff
@@ -146,10 +146,10 @@ def test_authenticated_redirect_studio_user(self, user_objects_manager):
...
-            'custom_assignment_id': 'irrelevant',
-            'custom_question_id': 1
+            'custom_assignment_id': 'assignment',
+            'custom_question_id': 1234
         }
-        expected_redirect = "/admin_index_wrapper/"
+        expected_redirect = "/assignment/assignment/1234/"
         actual_redirect = self.manager.authenticated_redirect_to(request, lti_data)
 
         self.assertEqual(actual_redirect, expected_redirect)
```
Comment (pomegranited) [inline dalite/tests.py]: Please also add tests for the new context property to `peerinst/tests/test_views.py`, e.g.
```python
class StaffUserContextProcessorTestCase(QuestionViewTestCase):

    def setUp(self):
        super(StaffUserContextProcessorTestCase, self).setUp()
        self.staff_user = factories.UserFactory()
        self.staff_user.is_staff = True
        self.staff_user.save()

    def test_student_user(self):
        """Test student context"""
        self.log_in_with_lti()
        response = self.question_get()
        self.assertFalse(response.context['user'].is_staff)
        self.assertFalse(response.context['is_staff_user_from_lms'])

    def test_staff_user_in_lms(self):
        """Test staff view from LMS"""
        self.log_in_with_lti(user=self.staff_user)
        response = self.question_get()
        self.assertTrue(response.context['user'].is_staff)
        self.assertFalse(response.context['is_staff_user_from_lms'])

    def test_staff_user_in_studio(self):
        """Test staff view from Studio"""
        # Staff users in Studio have username "student"
        self.staff_user.username = "student"
        self.staff_user.save()

        self.log_in_with_lti(user=self.staff_user)
        response = self.question_get()
        self.assertTrue(response.context['user'].is_staff)
        self.assertTrue(response.context['is_staff_user_from_lms'])
```

File: peerinst/templates/peerinst/question_list.html (hunk 1 above)
Comment (pomegranited) [inline peerinst/templates/peerinst/question_list.html]: Thank you for spotting that!

File: peerinst/templates/peerinst/question_start.html [hunk 4]
```diff
@@ -39,6 +39,13 @@
     </form>
     {% endblock %}
     {% endif %}
+
+  {% if is_staff_user_from_lms %}
+  <div class="admin-links">
+    <a class="light-button" href="{% url "admin:peerinst_question_change" question.id %}" target="_blank">{% trans "Edit this question" %}</a>
```
Comment (pomegranited) [inline peerinst/templates/peerinst/question_start.html]: Nice to have both the "view admin" button and this "edit question" button, thank you for adding both!

**Comment by pomegranited** [comment]
@jbzdak I've manually tested this change, and it meets the acceptance criteria laid out in OC-1643.

I've suggested renaming the context processor and property to `is_staff_user_from_studio`, and adding a few tests for that property, but other than that, the change looks good.

I'd like to re-run the automated tests once these changes are made, but other than that, everything looks good!

File: dalite/tests.py (hunk 3 above)
Comment (jbzdak) [inline dalite/tests.py]: This is now moot, since decision where to redirect used is made basing on action parameter.

**Comment by pomegranited** [comment]
@jbzdak My testing notes are on the [xblock-dalite PR #3](https://github.com/open-craft/xblock-dalite/pull/3#issuecomment-222056716)

This change looks good 👍 

Noting this possible improvement for another task, if our instructors report any trouble using Dalite:

1. The Admin area doesn't make it easy to see the question IDs, nor does it make it obvious which bit of text is the assignment ID, so it's hard to see how to configure the XBlock or LTI course module.

**Comment by jbzdak** [comment]
Squashed, re-tested, merging. 

@pomegranited Thanks for the fast review. I agree that, in terms of usability, we should find some way to enable staff users to set `question_id` and `assignment_id` more easily. However the only sensible and robust way, I could think of, would involve adding rest API to the dalite (to problematically query questions) and is out of scope for this task.
